File: Domain/Training.cs
namespace Domain
{
    public class Training
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string City { get; set; }
        public string Venue { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<TrainingAttendee> Attendees { get; set; } = new List<TrainingAttendee>(); //we are adding collection of attendees to prevent assigning null to Attendees in responses, so we are initializing a List
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: client_app/src/app/models/training.ts
import { Profile } from "./profile";

export interface ITraining {
  id: string;
  title: string;
  description: string;
  category: string;
  date: Date | null;
  city: string;
  venue: string;
  hostUsername?: string;
  isCancelled?: boolean;
  isGoing?: boolean;
  isHost?: boolean;
  attendees: Profile[];
  host?: Profile;
}

export class Training implements ITraining {
  constructor(init: TrainingFormValues) {
    this.id = init.id!;
    this.title = init.title;
    this.date = init.date;
    this.description = init.description;
    this.category = init.category;
    this.city = init.city;
    this.venue = init.venue;
  }
  id: string;
  title: string;
  description: string;
  category: string;
  date: Date | null;
  city: string;
  venue: string;
  hostUsername?: string;
  isCancelled?: boolean;
  isGoing?: boolean;
  isHost?: boolean;
  attendees: Profile[] = [];
  host?: Profile;
}

export class TrainingFormValues {
  id?: string = undefined;
  title: string = "";
  category: string = "";
  description: string = "";
  date: Date | null = null;
  city: string = "";
  venue: string = "";

  constructor(Training?: TrainingFormValues) {
    if (Training) {
      this.id = Training.id;
      this.title = Training.title;
      this.description = Training.description;
      this.category = Training.category;
      this.date = Training.date;
      this.venue = Training.venue;
      this.city = Training.city;
    }
  }
}
